### PR TITLE
Update Shell and partially re-implement toolpathing

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: algorithm/cgal/ComputeToolpath.h [hunk 1]
```diff
@@ -1,25 +1,32 @@
 #pragma once
 
+#define VERBOSE
```
Comment (BarbourSmith) [inline algorithm/cgal/ComputeToolpath.h:3]: Is this meant to be left in?

File: algorithm/cgal/ComputeToolpath.h [hunk 2]
```diff
@@ -79,60 +92,112 @@ int ComputeToolpath(Geometry* geometry, size_t material_start,
...
 
+  const double kCellCutCost = 1;
+
+  auto ComputeCellCutCost = [&](Coord coord, double& cost) -> bool {
+    Cell* cell;
+    if (!GetCell(coord, cell) || cell->is_cut) {
+      cost = 0;
+      return true;
+    }
+    if (cell->is_protected) {
```
Comment (BarbourSmith) [inline algorithm/cgal/ComputeToolpath.h:103]: Protected cells are for things that the tool can't pass through?

File: algorithm/cgal/ComputeToolpath.h [hunk 3]
```diff
@@ -186,38 +293,71 @@ int ComputeToolpath(Geometry* geometry, size_t material_start,
...
+        // Remove it from consideration.
+        cell->is_visited = true;    // We can't move the tool through it.
+        cell->is_too_close = true;  // Too close to a protected cell.
+        cell->may_cut = false;
+        return;
+      }
+      if (last_cell) {
+        double distance = ComputeCoordDistance(last_cell->coord, coord);
+        if (distance < 1.9) {
+          // This cut is a 90 degree turn -- penalize it to reflect
```
Comment (BarbourSmith) [inline algorithm/cgal/ComputeToolpath.h:318]: 👍

**Review by BarbourSmith — APPROVED** [review]
Looks great to me :)

File: algorithm/cgal/ComputeToolpath.h (hunk 1 above)
Comment (pentacular) [reply algorithm/cgal/ComputeToolpath.h:3]: Yeah, it's a bit of a WIP.

**Pull request merged**